include/nil/crypto3/multiprecision/cpp_int_modular.hpp

```diff
@@ -0,0 +1,807 @@
+            public:
+                using local_limb_type = typename trivial_limb_type<Bits>::type;
+                using limb_pointer = local_limb_type*;
+                using const_limb_pointer = const local_limb_type*;
+                static constexpr unsigned limb_bits = sizeof(local_limb_type) * CHAR_BIT;
+                static constexpr limb_type max_limb_value = ~static_cast<limb_type>(0u);
+ 
+
+                struct scoped_shared_storage {
+                    constexpr scoped_shared_storage(const cpp_int_modular_base&, unsigned) {
+                    }
+                    constexpr void deallocate(unsigned) {
+                    }
+                };
+
+            private:
+                static constexpr local_limb_type limb_mask =
+                    limb_bits != Bits ?
+                        static_cast<local_limb_type>(static_cast<local_limb_type>(~local_limb_type(0)) >>
+                                                     (limb_bits - Bits)) :
+                        static_cast<local_limb_type>(~local_limb_type(0));
+
+                local_limb_type m_data;
+                bool m_carry = false;
+
+                //
+                // Interface invariants:
+                //
+                static_assert(Bits <= sizeof(double_limb_type) * CHAR_BIT,
+                              "Template parameter Bits is inconsistent with the parameter trivial - did you "
+                              "mistakingly try to override the trivial parameter?");
+
+            public:
+                //
+                // Direct construction:
+                //
+                template<class UI>
+                inline constexpr cpp_int_modular_base(
+                    UI i,
+                    typename std::enable_if<boost::multiprecision::detail::is_unsigned<UI>::value
+                                            >::type const* = 0) noexcept :
+                    m_data(static_cast<local_limb_type>(i) & limb_mask) {
+                }
+                constexpr cpp_int_modular_base(boost::multiprecision::literals::detail::value_pack<>) noexcept :
+                    m_data(static_cast<local_limb_type>(0u)) {
+                }
+                template<limb_type a>
+                constexpr cpp_int_modular_base(boost::multiprecision::literals::detail::value_pack<a>) noexcept :
+                    m_data(static_cast<local_limb_type>(a)) {
+                }
+                template<limb_type a, limb_type b>
+                constexpr cpp_int_modular_base(boost::multiprecision::literals::detail::value_pack<a, b>) noexcept :
+                    m_data(static_cast<local_limb_type>(a) | (static_cast<local_limb_type>(b) << bits_per_limb)) {
+                }
+                //
+                // These are deprecated in C++20 unless we make them explicit:
+                //
+                constexpr cpp_int_modular_base& operator=(const cpp_int_modular_base&) = default;
```

Comment:
They just mean that if you delete this line, before C++20 it would still get generated, but after C++20 you need to write "blablabla = default" to make it generate the default operator=.